src/main/app/claims/converters/claimantResponseConverter.ts

```diff
@@ -0,0 +1,58 @@
+import { DraftClaimantResponse } from 'features/claimant-response/draft/draftClaimantResponse.ts'
+import { ResponseRejection } from 'claims/models/response/core/responseRejection.ts'
+import { FormaliseRepaymentPlanOption } from 'claimant-response/form/models/formaliseRepaymentPlanOption'
+import { CourtDetermination } from 'claims/models/response/core/courtDetermination.ts'
+import { PaymentIntention } from 'claims/models/response/core/paymentIntention'
+
+export class ClaimantResponseConverter {
+
+  public static covertToClaimantRespose (draftClaimantResponse: DraftClaimantResponse): any {
+
+    if (draftClaimantResponse.rejectionReason && draftClaimantResponse.rejectionReason.text.length > 0) {
+      return {
+        'type': 'rejection',
+        'amountPaid': 0,
+        'responseRejection': this.createResponseRejection(draftClaimantResponse)
+      }
+    } else {
+      return {
+        'type': 'acceptation',
+        'amountPaid': draftClaimantResponse.paidAmount ? draftClaimantResponse.paidAmount : 0,
+        'courtDetermination': this.createCourtDetermination(draftClaimantResponse),
+        'formaliseOption': this.getFormaliseOption(draftClaimantResponse.formaliseRepaymentPlan.option)
+      }
+    }
+  }
+
+  public static createResponseRejection (draftClaimantResponse: DraftClaimantResponse): ResponseRejection {
+    return {
+      freeMediation: draftClaimantResponse.freeMediation.option === 'yes',
+      reason: draftClaimantResponse.rejectionReason.text
+    } as ResponseRejection
+  }
+
+  public static createCourtDetermination (draftClaimantResponse: DraftClaimantResponse): CourtDetermination {
+    return {
+      'courtCalculatedPaymentIntention': this.createPaymentIntention(draftClaimantResponse.courtOfferedPaymentIntention), // TODO:completion Date is missing in the FE model
+      'rejectionReason': draftClaimantResponse.rejectionReason ? draftClaimantResponse.rejectionReason.text : ''
+    }
+  }
+
+  public static getFormaliseOption (option: FormaliseRepaymentPlanOption): string {
+    switch (option) {
+      case FormaliseRepaymentPlanOption.SIGN_SETTLEMENT_AGREEMENT :
+        return 'SETTLEMENT'
+      case FormaliseRepaymentPlanOption.REQUEST_COUNTY_COURT_JUDGEMENT :
+        return 'CCJ'
+    }
+  }
+
+  public static createPaymentIntention (paymentIntention: PaymentIntention): any {
```

Comment:
Firstly that should be already implemented somewhere as this is not the first time we serialise payment intention instances.

Secondarily please do not use any in type definitions. Any is allowed only when you really struggle to make something work.

Reply:
I did indeed see  a function convertPaymentIntention()  in responseModelConverter.ts but that was not reusable in my case. 

I agree with you 'any' is not the right type to use here. There is already src/main/app/claims/models/response/core/paymentIntention.ts but that can not be reused here.. I will create a new paymentIntention  interface , if it is ok with you ?